OpenRA.Mods.Common/Traits/Render/WithDeathAnimation.cs

```diff
@@ -71,10 +74,23 @@ public WithDeathAnimation(Actor self, WithDeathAnimationInfo info)
 
 		public void Killed(Actor self, AttackInfo e)
 		{
-			// Killed by some non-standard means. This includes being crushed
-			// by a vehicle (Actors with Crushable trait will spawn CrushedSequence instead).
+			// Actors with Crushable trait will spawn CrushedSequence.
+			var crushable = self.TraitOrDefault<Crushable>();
+			if (crushable != null && crushable.Crushed)
```

Comment:
Unfortunately this (nice in theory) solution won't work out:
If I use this approach, I'd need to have `ICrushable.CrushableBy` `return true` here,
as otherwise it would block out every crush attempt. But when I `return true` and `WithDeathAnimation` is the only trait that implements `ICrushable` on the actor, the actor can be "crushed" by everything suddenly